cmd/werf/common/telemetry.go

```diff
@@ -120,21 +120,41 @@ func getTelemetryProjectID(ctx context.Context) (string, error) {
 			return "", fmt.Errorf("unable to get repo origin url: %w", err)
 		}
 
-		ep, err := transport.NewEndpoint(url)
+		hash, err := hashOriginUrl(url)
 		if err != nil {
-			return "", fmt.Errorf("bad repo origin url %q: %w", url, err)
+			return "", fmt.Errorf("unable to hash origin url: %w", err)
 		}
 
-		hashParts := []string{ep.Protocol, ep.Host, fmt.Sprintf("%d", ep.Port), ep.Path}
-
 		telemetry.LogF("calculate projectID based on repo origin url")
-
-		projectID = util.Sha256Hash(hashParts...)
+		projectID = hash
 	}
 
 	return projectID, nil
 }
 
+func hashOriginUrl(url string) (string, error) {
+	ep, err := transport.NewEndpoint(url)
+	if err != nil {
+		return "", fmt.Errorf("bad repo origin url %q: %w", url, err)
+	}
+
+	var formatPath string
+	{
+		formatPath = ep.Path
+
+		paramsIndex := strings.Index(formatPath, "?")
```

Comment:
Could there be the `?` symbol somewhere before the GET parameters? I would've Split the `ep.Path` by `/` and would've checked only the last part for `?` occurence.

Reply:
The parameters can include `/` (e.g. `?path=dir/file.txt`) . And I am not sure about the possibility of using `?` in the rest of path.